sda-commons-server-kms-aws/src/test/java/org/sdase/commons/server/kms/aws/util/MockKMSClient.java

```diff
@@ -0,0 +1,367 @@
+    throw new java.lang.UnsupportedOperationException();
+  }
+
+  @Override
+  public RetireGrantResult retireGrant(RetireGrantRequest arg0) throws AmazonClientException {
+    throw new java.lang.UnsupportedOperationException();
+  }
+
+  @Override
+  public RevokeGrantResult revokeGrant(RevokeGrantRequest arg0) throws AmazonClientException {
+    throw new java.lang.UnsupportedOperationException();
+  }
+
+  @Override
+  public void setEndpoint(String arg0) {
+    // Do nothing
+  }
+
+  @Override
+  public void setRegion(Region arg0) {
+    region_ = arg0;
+  }
+
+  @Override
+  public void shutdown() {
+    // Do nothing
+  }
+
+  @Override
+  public UpdateKeyDescriptionResult updateKeyDescription(UpdateKeyDescriptionRequest arg0)
+      throws AmazonClientException {
+    throw new java.lang.UnsupportedOperationException();
+  }
+
+  public void deleteKey(final String keyId) {
+    final String arn = retrieveArn(keyId);
+    activeKeys.remove(arn);
+  }
+
+  private String retrieveArn(final String keyId) {
+    String arn = keyAliases.get(keyId);
+    assertExists(arn);
+    return arn;
+  }
+
+  private void assertExists(String keyId) {
+    if (keyAliases.containsKey(keyId)) {
+      keyId = keyAliases.get(keyId);
+    }
+    if (keyId == null || !activeKeys.contains(keyId)) {
+      throw new NotFoundException("Key doesn't exist: " + keyId);
+    }
+  }
+
+  private static class DecryptMapKey {
+    private final ByteBuffer cipherText;
+    private final Map<String, String> ec;
+
+    public DecryptMapKey(DecryptRequest req) {
+      cipherText = req.getCiphertextBlob().asReadOnlyBuffer();
+      if (req.getEncryptionContext() != null) {
+        ec = Collections.unmodifiableMap(new HashMap<String, String>(req.getEncryptionContext()));
+      } else {
+        ec = Collections.emptyMap();
+      }
+    }
+
+    public DecryptMapKey(ByteBuffer ctBuff, Map<String, String> ec) {
+      cipherText = ctBuff.asReadOnlyBuffer();
+      if (ec != null) {
+        this.ec = Collections.unmodifiableMap(new HashMap<String, String>(ec));
+      } else {
+        this.ec = Collections.emptyMap();
+      }
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = 1;
+      result = prime * result + ((cipherText == null) ? 0 : cipherText.hashCode());
+      result = prime * result + ((ec == null) ? 0 : ec.hashCode());
+      return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) return true;
+      if (obj == null) return false;
+      if (getClass() != obj.getClass()) return false;
+      DecryptMapKey other = (DecryptMapKey) obj;
+      if (cipherText == null) {
+        if (other.cipherText != null) return false;
+      } else if (!cipherText.equals(other.cipherText)) return false;
+      if (ec == null) {
+        if (other.ec != null) return false;
+      } else if (!ec.equals(other.ec)) return false;
```

Comment:
Arg, please don't ever ever use if statements without code blocks. Adding a single line her can cause havoc. We even [have a rule for that](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used). I guess the code can be simplified a bit.